### PR TITLE
[docker-ptf]: CI change to publish docker-ptf image to the docker registry

### PR DESCRIPTION
**Pull request by opcoder0**
#### Why I did it

Changes to `docker-ptf` Dockerfile don't get automatically published to the docker registry accessed by the community. This PR addresses the issue.

##### Work item tracking
- Microsoft ADO **(number only)**: 29869315

#### How I did it

Modified the `postSteps` phase in `azure-pipelines-image-template.yml` to invoke the script to push `docker-ptf` image to the docker registry. The push is applicable to build branches (`master`, `202305`, `202311` etc.) and not feature branches.

#### How to verify it

TBD

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

Not applicable.

#### Description for the changelog

[docker-ptf]: CI change to publish docker-ptf image to the public/community docker registry

* Change pipeline YAML to publish docker-ptf image to docker registry
* Publish only docker-ptf image for main or release branches only and not feature branch

#### Link to config_db schema for YANG module changes

Not applicable

#### A picture of a cute animal (not mandatory but encouraged)

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -52,6 +52,8 @@ variables:
 stages:
 - stage: BuildVS
   pool: sonicbld-1es
+  variables:
```
Comment (liushilongbuaa) [inline azure-pipelines.yml:55]: azure-pipelines.yml don't need this Variable Group
official-build.yml need this.

File: .azure-pipelines/azure-pipelines-image-template.yml [hunk 2]
```diff
@@ -58,9 +58,23 @@ jobs:
         displayName: 'Make configure'
     postSteps:
       - script: |
+          BUILD_REASON=$(Build.Reason)
+          if [[ "$BUILD_REASON" != "PullRequest" && "$GROUP_NAME" == "vs" ]]
```
Comment (liushilongbuaa) [inline .azure-pipelines/azure-pipelines-image-template.yml:62]: Azure.sonic-buildimage.official.vs
Please use pipeline name in condition. Only this pipeline need the step.

File: .azure-pipelines/azure-pipelines-image-template.yml [hunk 3]
```diff
@@ -58,9 +58,22 @@ jobs:
...
+            [[ "$BRANCH" == "master" ]] && LABELS="$LABELS latest"
+            for f in $DOCKERS; do
+              echo $f
+              ./push_docker.sh $f $(REGISTRY_SERVER_PUBLIC) $PORT $(REGISTRY_USERNAME) "$REGISTRY_PASSWD" "$LABELS"
+            done
+          fi
           mkdir -p $(Build.ArtifactStagingDirectory)/target
           mv target/* $(Build.ArtifactStagingDirectory)/target/
-        displayName: Copy Artifacts
+        displayName: Publish to Docker Registry and Copy Artifacts
```
Comment (liushilongbuaa) [inline .azure-pipelines/azure-pipelines-image-template.yml:76]: you need add something like this
env:
    REGISTRY_PASSWD: $(ABC)

**Comment by opcoder0** [comment]
@liushilongbuaa I have tested the pipeline and the publish is working. Can you please take another look.

**Review by wangxin — APPROVED** [review]

**Review by liushilongbuaa — APPROVED** [review]

**Pull request merged**